src/coreclr/gc/gc.cpp

```diff
@@ -9349,6 +9367,60 @@ size_t gc_heap::sort_mark_list()
     }
 #endif //USE_VXSORT
 
+    uint8_t** x = mark_list;
+
+#ifdef USE_REGIONS
+    // first set the pieces for all regions to empty
+    size_t lowest_region_index = (size_t)g_gc_lowest_address >> gc_heap::min_segment_size_shr;
+    size_t highest_region_index = (size_t)g_gc_highest_address >> gc_heap::min_segment_size_shr;
+    size_t region_count = highest_region_index - lowest_region_index + 1;
+    if (mark_list_piece_size < region_count)
+    {
+        delete[] mark_list_piece_start;
+        delete[] mark_list_piece_end;
+
+        mark_list_piece_start = new (nothrow) uint8_t** [region_count];
+        mark_list_piece_end   = new (nothrow) uint8_t** [region_count];
```

Comment:
this is definitely not urgent (because we don't even change the lowest/highest address right now) but in the future when we do we'd want some kind of growth factor instead of allocating the exact region_count elements...basically like what we do with all the other data structures when we grow.

Reply:
Makes sense - done.